ui/src/reportAnIssue.ts

```diff
@@ -33,22 +41,30 @@ Product: ${vscode.env.appName}
 Product Version: ${vscode.version}
 Language: ${vscode.env.language}`;
 
-    if (parsedError.stack) {
-        body = body.concat(`
+    // Try repeatedly with fewer lines of stack until we have a URL of a reasonable length
+    // tslint:disable-next-line: no-increment-decrement
+    for (let stackLines: number = numberOfLines(stack); stackLines >= 0; --stackLines) {
+        let bodyWithStack: string = body;
+        if (stack) {
```

Comment:
```
if (stack && stackLines > 0) {
```

No need to add extra characters if we can avoid it

Reply:
if stack = "", stack is falsy

Reply:
No I'm talking about the case where the body by itself is close to 2000 characters. Aka the stack has contents and you've already tried a bunch of times to add some of it's lines